### ISSUE TITLE
Reference to objects are not supported?

### ISSUE THREAD

**Issue by YukinoHayakawa** [jonigata/squall#9]
Seems that neither Fetch nor push_aux takes that into consideration🙁🙁

**Comment by jonigata** [comment]
I guess supporting reference is somewhat danger because of its lifetime. Do you think we should consider reference to objects?

**Comment by YukinoHayakawa** [comment]
Ideally it should handle object -> reference conversion transparently.

I just tried a function returning an object and another accepting one, it also caused access violation for some reason. I think as Sq uses reference counting to track objects maybe it would be better to use shared_ptr to manage objects transferred to the script?

**Comment by jonigata** [comment]
I thought that if user wants managing memory strictly, he should manage all instances' lifetime inside c++, and refer them from squirrel script. Do you want to use squirrel GC to manage c++ objects?

**Comment by YukinoHayakawa** [comment]
Yes. I plan to have some shared objects whose only strong ref is from sq.

**Comment by jonigata** [comment]
Previously I have exchanged mails with anothor user(A).

the mails were:

A:
I'm currently using your quall project at github and I'm trying to bind an c++ object, using the sample as a template.

It works as it, if I call from C++ a squirrel function and pass as a parameter the object, but the problem is that I can't instanciate a new object in the squirrel code.

You know if it is posible to do that using squall?

Me:
There is one thing I can say that instantiating C++ object in squirrel brings 
a bit complex issue.
1.If you want to manage memory by yourself, I think you have better to write
static function that instantiates a C++ object.
2.If you want to use squirrel GC, I guess I have to extend squall by defining 
constructor binding, because it requires custom deleter.

**Comment by jonigata** [comment]
Is your requirements like her?

**Comment by YukinoHayakawa** [comment]
Not exactly, I create objects from a C++ factory function and use it in sq.
